packages/samples/headless-react/src/pages/PageWithHooks.tsx

```diff
@@ -0,0 +1,36 @@
+import {
+  buildFacet,
+  buildSearchEngine,
+  getSampleSearchEngineConfiguration,
+} from '@coveo/headless';
+import {useEffect} from 'react';
+import {
+  FacetWithGenericDecorator,
+  FacetWithSpecificDecorator,
+} from '../components/facet/facet.class.decorator';
+import {FacetWithHook} from '../components/facet/facet.fn.hooks';
+
+import {AppContext} from '../context/engine';
+import {Section} from '../layout/section';
+
+export function PageWithHooks() {
```

Comment:
I see this component being part of the package api, either called `SearchEngineProvider` or `SearchProvider` to align with React conventions.

The render function would only set the provider and render children passed to it.

```
return (
  <AppContext.Provider value={{engine}}>
      {props.children()}
  </AppContext.Provider>
)
```

Example usage:

```
return(
    <SearchEngineProvider configuration={...}>
        <SearchBox/>
        <ResultList/>
    </SearchEngineProvider>
)
```

Reply:
Yes, a context provider should be part of the package !

Reply:
Like in the headless + react cli! I found that very helpful!